### PR TITLE
[TACHYON-1582] Restructure POM: move servers/client/common into core

### PR DESCRIPTION
**Pull request by apc999**
https://tachyon.atlassian.net/browse/TACHYON-1582
This is the followup PR to achieve what proposed in https://tachyon.atlassian.net/secure/attachment/14000/POMstructurechangeproposal.pdf

NOTE that, this PR has touched a large number of files and hitting the github limitation, but most of the files are simply renamed. There is a small amount of code change into POM files. If you want to see the whole change, please check out the branch.

### CODE REVIEW

**Comment by apc999** [comment]
An alternative way of reviewing this PR is to review it by commits, keeping in mind that the first commit simply move files. Most pom changes happen in https://github.com/apc999/tachyon/commit/815df11b0de3da6455716ae7476cd6af7601c3a0

**Comment by aaudiber** [comment]
I think we should rename `tachyon-client-core` and `tachyon-client-keyvalue` to `tachyon-core-client` and `tachyon-keyvalue-client`, and I left a comment about some comments, but otherwise LGTM. Thanks for this reorganization, I like the new structure much better!

**Comment by aaudiber** [comment]
LGTM

File: bin/tachyon [hunk 1]
```diff
@@ -305,9 +305,10 @@ case ${COMMAND} in
   exit 0
 ;;
 "protoGen")
-  rm -rf ${BIN}/../servers/src/main/java/tachyon/proto
-  for src_file in $(find ${BIN}/../servers/src/proto -type f); do
-    protoc --java_out=${BIN}/../servers/src/main/java --proto_path=`dirname ${src_file}` ${src_file}
+  rm -rf ${BIN}/../core/server/src/main/java/tachyon/proto
+  for src_file in $(find ${BIN}/../core/server/src/proto -type f); do
+    protoc --java_out=${BIN}/../core/server/src/main/java --proto_path=`dirname ${src_file}`
+    ${src_file}
```
Comment (uronce-cc) [inline bin/tachyon]: Why would this be on a new line?

**Comment by calvinjia** [comment]
LGTM

File: bin/tachyon [hunk 2]
```diff
@@ -305,9 +305,10 @@ case ${COMMAND} in
   exit 0
 ;;
 "protoGen")
```
Comment (uronce-cc) [inline bin/tachyon]: Also, thriftGen needs to be updated.

**Comment by uronce-cc** [comment]
@apc999 Left two comments.

File: bin/tachyon (hunk 2 above)
Comment (apc999) [inline bin/tachyon]: fixed

**Comment by apc999** [comment]
@uronce-cc  your comments are addressed. Thanks!

File: bin/tachyon (hunk 1 above)
Comment (apc999) [inline bin/tachyon]: removed

**Comment by uronce-cc** [comment]
LGTM ++

**Comment by apc999** [comment]
Jenkins, test this please

**Comment by apc999** [comment]
Jenkins, test this please

**Pull request merged**